drltrace/drltrace_frontend.cpp

```diff
@@ -0,0 +1,314 @@
+/* ***************************************************************************
+ * Copyright (c) 2017 Google, Inc.  All rights reserved.
+ * ***************************************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* front end for drltrace tool */
+
+#ifdef WINDOWS
+# define UNICODE
+# define _UNICODE
+#endif
+
+#include "dr_api.h"
+#include "dr_inject.h"
+#include "dr_config.h"
+#include "dr_frontend.h"
+#include "droption.h"
+#undef TESTANY
+#include "utils.h"
+#include <string.h>
+
+#define MAX_DR_CMDLINE (MAXIMUM_PATH*6)
+
+#define DRLTRACE_ERROR(msg, ...) do { \
+    fprintf(stderr, "ERROR: " msg "\n", ##__VA_ARGS__);    \
+    fflush(stderr); \
+    exit(1); \
+} while (0)
+
+static droption_t<std::string> op_logdir
+(DROPTION_SCOPE_ALL, "logdir", ".", "Log directory to print library call data",
+ "Specify log directory where library call data will be written, in a separate file per \
+process. The default value is \".\" (current dir) If set to \"-\", data for all \
+processes are printed to stderr (warning: this can be slow).");
+static droption_t<bool> op_only_from_app
+(DROPTION_SCOPE_CLIENT, "only_from_app", false, "Reports only library calls from the app",
+ "Only reports library calls from the application itself, as opposed to all calls even \
+from other libraries or within the same library.");
+static droption_t<bool> op_no_follow_children
```

Comment:
I didn't understand that :) If I set op_follow_children to be true by default how can user set it in false ?